packages/nc-gui-v2/utils/validation.ts

```diff
@@ -83,7 +83,39 @@ export const fieldRequiredValidator = {
   required: true,
   message: 'Field is required',
 }
+
 export const getRequiredValidator = (field = 'Field') => ({
   required: true,
   message: `${field} is required`,
 })
+
+export const importUrlValidator = {
+  validator: (rule: any, value: any, callback: (errMsg?: string) => void) => {
+    if (
+      /(10)(\.([2]([0-5][0-5]|[01234][6-9])|[1][0-9][0-9]|[1-9][0-9]|[0-9])){3}|(172)\.(1[6-9]|2[0-9]|3[0-1])(\.(2[0-4][0-9]|25[0-5]|[1][0-9][0-9]|[1-9][0-9]|[0-9])){2}|(192)\.(168)(\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])){2}|(0.0.0.0)|localhost?/g.test(
+        value,
+      )
+    ) {
+      callback('IP Not allowed!')
+    }
+    callback()
+  },
+}
+
+export const importCsvUrlValidator = {
+  validator: (rule: any, value: any, callback: (errMsg?: string) => void) => {
+    if (value && !/.*\.(csv)/.test(value)) {
+      callback('Target file is not an accepted file type. The accepted file type is .csv!')
+    }
+    callback()
+  },
+}
+
+export const importExcelUrlValidator = {
+  validator: (rule: any, value: any, callback: (errMsg?: string) => void) => {
+    if (value && !/.*\.(xls|xlsx|xlsm|ods|ots)/.test(value)) {
+      callback('Target file is not an accepted file type. The accepted file types are .xls, .xlsx, .xlsm, .ods, .ots!')
+    }
+    callback()
+  },
```

Comment:
Same as above, callback is deprecated and throws error/warnings in console, so use a Promise return instead